nautobot/docs/development/docker-compose-advanced-use-cases.md

```diff
@@ -165,3 +165,132 @@ $ docker-compose -f docker-compose.yml -f docker-compose.debug.yml up
+
+    Copy the file `invoke.yml.example` to `invoke.yml` and configure at least the `compose_dir` according your project location.
+
+3. **Configure `docker-compose.override.yml`**
+
+    It is important not to use the same listen port for the `nautobot` and `celery_worker` services.
+
+    Unlike the current `celery_worker` the container will not automatically restart on source changes.
+    But it is possible to restart the container from the VS Code Docker extension.
+
+    ```yaml
+    # We can't remove volumes in a compose override, for the test configuration using the final containers
+    # we don't want the volumes so this is the default override file to add the volumes in the dev case
+    # any override will need to include these volumes to use them.
+    # see:  https://github.com/docker/compose/issues/3729
+    ---
+      version: "3"
+      services:
+        nautobot:
+          command: python /tmp/debugpy --listen 0.0.0.0:6899 -m manage runserver 0.0.0.0:8080 --insecure
+          ports:
+            - "8080:8080"
+            - "6899:6899"
+          volumes:
+            - ./nautobot_config.py:/opt/nautobot/nautobot_config.py
+            - ../:/source
+        celery_worker:
+          entrypoint: python /tmp/debugpy --listen 0.0.0.0:6898 -m manage celery worker -l DEBUG --pool=solo --events
+          ports:
+            - "6898:6898"
+          volumes:
+            - ./nautobot_config.py:/opt/nautobot/nautobot_config.py
+            - ../:/source
+        celery_beat:
+          volumes:
+            - ./nautobot_config.py:/opt/nautobot/nautobot_config.py
+            - ../:/source
+    ```
+
+4. **Rebuild the container**
+
+    ```bash
+    $ invoke build [--no-cache]
+    ```
+
+5. **Add the debug configuration to VS Code**
+
+    Add the following debug configuration to `.vscode/launch.json`:
+
+    ```json
+    {
+      // Use IntelliSense to learn about possible attributes.
+      // Hover to view descriptions of existing attributes.
+      // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+      "version": "0.2.0",
+      "configurations": [
+        {
+          "name": "Nautobot REMOTE",
+          "type": "python",
+          "request": "attach",
+          "connect": {
+            "host": "127.0.0.1",
+            "port": 6899
+          },
+          "pathMappings": [
+            {
+              "localRoot": "${workspaceFolder}",
+              "remoteRoot": "/source"
+            }
+          ],
+          "django": true
+        },
+        {
+          "name": "Nautobot-CELERY REMOTE",
+          "type": "python",
+          "request": "attach",
+          "connect": {
+            "host": "127.0.0.1",
+            "port": 6898
+          },
+          "pathMappings": [
+            {
+              "localRoot": "${workspaceFolder}",
+              "remoteRoot": "/source"
+            }
+          ],
+          "django": true
+        }
```

Comment:
These can be added to the `launch` section in `nautobot.code-workspace`

Reply:
Done